packages/console/src/ConsoleInput.jsx

```diff
@@ -144,29 +144,33 @@ export class ConsoleInput extends PureComponent {
      * Can't do it in `onDidChangeModelContent` either, since we want to stop the Enter action from modifying the command.
      */
     this.commandEditor.onKeyDown(keyEvent => {
-      if (
-        !this.isCommandModified ||
-        this.commandEditor.getModel().getValueLength() === 0
-      ) {
-        if (keyEvent.keyCode === monaco.KeyCode.UpArrow) {
-          const { commandHistoryIndex } = this;
+      const { commandEditor, commandHistoryIndex } = this;
+      const { lineNumber } = commandEditor.getPosition();
+      const model = commandEditor.getModel();
+      if (!this.isCommandModified || model.getValueLength() === 0) {
```

Comment:
I don't think we want this if check anymore? Chrome doesn't do it, and you can get into a state where you can't cycle through commands again until you fully delete a command you've been editing, and then it's still in the point in history you were at... probably better to just remove this check now that you have the correct top/bottom behaviour